pkg/tls/tls.go

```diff
@@ -1,91 +1,164 @@
 package tls
 
 import (
-	"bytes"
-	"crypto/ecdsa"
-	"crypto/elliptic"
-	"crypto/rand"
 	"crypto/tls"
 	"crypto/x509"
-	"crypto/x509/pkix"
-	"encoding/pem"
-	"math/big"
-	"net"
-	"time"
+	"errors"
+	"io/ioutil"
+	"os"
 )
 
-// GenerateTLSConfig Setup a bare-bones TLS config for the server
-func GenerateTLSConfig(host ...string) *tls.Config {
-	tlsCert, _ := GenerateCertificate(host...)
-
-	return &tls.Config{
-		Certificates:       []tls.Certificate{tlsCert},
-		ClientSessionCache: tls.NewLRUClientSessionCache(1),
-		NextProtos:         []string{"yomo"},
+// Create server tls config.
+func CreateServerTLSConfig() (*tls.Config, error) {
```

Comment:
@wujunzhuo should change comment to `CreateServerTLSConfig create server tls config.`